instance/api/instance.py

```diff
@@ -51,9 +52,13 @@ def provision(self, request, pk=None):
             return Response({'status': 'Instance is not ready for reprovisioning'},
                             status=status.HTTP_403_FORBIDDEN)
 
-        instance.set_to_branch_tip()
-        provision_instance(pk)
+        try:
+            instance.set_to_branch_tip()
+        except github.ObjectDoesNotExist:
+            return Response({'status': "Ref '{0}' not found".format(instance.branch_name)},
```

Comment:
I think we can be a bit more specific here.  While probably any git ref would work, for auto-created sandboxes this will always be a branch name.  I'd also include the hint that the branch has probably been deleted.